COMP: Support ITK_LEGACY_SILENT for ThreadInfoStruct

N-Dekker opened the pull request:
Silenced "ThreadInfoStruct is deprecated" warning when the user has switched on
the ITK CMake option `ITK_LEGACY_SILENT`.

hjmjohnson commented:
Concept approved.  Of course the dashboard testing needs to pass :).

fbudin69500 commented:
Yes once https://github.com/InsightSoftwareConsortium/ITK/pull/698#pullrequestreview-224461901 is merged. I needed to merge #703 to test #698 correctly. Now I need to merge #698 for everything to work. I would not worry about failing CI before #698 is merged.

fbudin69500 commented:
@hjmjohnson Sorry about my previous message. It was actually #697 that needed to be merged (in `master`). #698 was the PR against `python-builds` that we created solely to run the Python tests on CI. I un-merged #698 from `python-builds` and merged `#697 into master.

thewtex commented:
@N-Dekker could this please be rebased for CI updates?

fbudin69500 commented:
@N-Dekker I rebased on my computer and tried to push on your branch, but this was not possible. I will let you try to rebase. Hopefully once it is rebased, CI will be green.

N-Dekker commented:
@ fbudin69500 
> @N-Dekker I rebased on my computer and tried to push on your branch, but this was not possible. I will let you try to rebase.

Thanks for trying, Francois! Sorry, I did not enable "edits from maintainers" for this pull request, because in general, I would like to keep control over the PRs that I submit. Hope that's OK to you.  `:-)`

Anyway, I just rebased, hoping to see a green CI!
